lib/elixir/lib/file.ex

```diff
@@ -730,6 +730,26 @@ defmodule File do
     :file.rename(source, destination)
   end
 
+  @doc """
+  The same as `rename/2` but raises a `File.RenameError` exception if it fails.
+  Returns `:ok` otherwise.
+  """
+
+  @spec rename!(Path.t(), Path.t()) :: :ok | {:error, posix}
+  def rename!(source, destination) do
+    case rename(source, destination) do
+      :ok ->
+        :ok
+
+      {:error, reason} ->
+        raise File.RenameError,
+          reason: reason,
+          action: "rename",
+          source: maybe_to_string(source),
```

Comment:
I think this can use `IO.chardata_to_string/1` since the function only accepts `Path.t()`

Reply:
Yes, it will be better. 
Fixed.